### PR TITLE
EOS-20611: Added wait for node online after cluster creation

### PR DESCRIPTION
**Pull request by indrajitzagade**
## Problem Statement
<pre>
  <code>
  Story Ref (if any):
    3-node VM deployment fails sometimes.
  </code>
</pre>
## Unit testing on RPM done
<pre>
  <code>
  Yes
  </code>
</pre>
## Problem Description
<pre>
  <code>
    3-node VM deployment fails sometimes since the node does not become online after the cluster creation.
  </code>
</pre>
## Solution
<pre>
  <code>
    Added a configurable wait for node to become online after the cluster creation.
  </code>
</pre>
## Unit Test Cases
<pre>
  <code>
    Tested on the setup where the issue was seen. Tried playing with the wait time to confirm the retry mechanism works fine. Tested both success and failure cases.
  </code>
</pre>

### CODE REVIEW

**Review by ajay-paratmandali — APPROVED** [review]

File: ha/core/controllers/pcs/cluster_controller.py [hunk 1]
```diff
@@ -52,6 +52,20 @@ def _is_pcs_cluster_running(self):
             return False
         return True
 
+    def _wait_for_node_online(self, nodeid: str) -> bool:
+        """
+        Wait till the node becomes online
+        """
+        for retry_index in range(0, const.CLUSTER_RETRY_COUNT):
```
Comment (ajaysrivas) [inline ha/core/controllers/pcs/cluster_controller.py:59]: Currently we are waiting for max 30 sec. - 5*6. Is it possible that this time increases with the number of nodes in the cluster?

Comment (indrajitzagade) [reply ha/core/controllers/pcs/cluster_controller.py:59]: I guess so @ajaysrivas.

Comment (ajaysrivas) [reply ha/core/controllers/pcs/cluster_controller.py:59]: Can we create a github issue or jira ticket to experiment and if required, use number of nodes in place of CLUSTER_RETRY_COUNT.

Comment (indrajitzagade) [reply ha/core/controllers/pcs/cluster_controller.py:59]: Created https://jts.seagate.com/browse/EOS-20642 @ajaysrivas

**Review by ajaysrivas — APPROVED** [review]

**Pull request merged**